linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/SourceMapWriter.scala

```diff
@@ -269,17 +269,30 @@ final class SourceMapWriter(out: ByteArrayWriter, jsFileName: String,
   protected def doWriteSegment(columnInGenerated: Int, pos: Position, name: String): Unit = {
     // scalastyle:off return
 
+    /* This method is incredibly performance-sensitive, so we resort to
+     * "unsafe" direct access to the underlying array of `out`.
+     */
+    val MaxSegmentLength = 1 + 5 * 7 // ',' + max 5 base64VLQ of max 7 bytes each
+    val buffer = out.unsafeStartDirectWrite(maxBytes = MaxSegmentLength)
```

Comment:
Have you considered using a `ByteBuffer` to track the offset here? (or does that destroy the performance gains)?

Reply:
The `ByteBuffer` suggestion kills the performance gains.